crates/phactory/src/system/mod.rs

```diff
@@ -678,23 +688,13 @@ impl<Platform: pal::Platform> System<Platform> {
                 }
             }
         }
+        self.process_contract_messages(block);
```

Comment:
Why do we move the contract message processing here? Is it called multiple times a block?

Reply:
Yes, It is called once per message rather than once per block.